### PR TITLE
[Psdp 358 ] allow add custom perspective element tree in element t…

### PR DESCRIPTION
**Pull request by samyemad**

### CODE REVIEW

**Comment by samyemad** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/Resources/public/js/pimcore/perspective/perspective.js [hunk 1]
```diff
@@ -674,33 +674,43 @@ pimcore.bundle.perspectiveeditor.PerspectiveEditor = class {
         if(!config.treeContextMenu){
             config.treeContextMenu = {};
         }
-
-        var treeElementTypStore = new Ext.data.Store({
+        let elementTreeTypeStoreId = 'perspective_editor_element_tree_type_store';
+        let elementTreeTypeStore = new Ext.data.Store({
```
Comment (robertSt7) [inline src/Resources/public/js/pimcore/perspective/perspective.js]:
```suggestion
        const elementTreeTypeStoreId = 'perspective_editor_element_tree_type_store';
        const elementTreeTypeStore = new Ext.data.Store({
```

File: src/Resources/public/js/pimcore/perspective/perspective.js [hunk 2]
```diff
@@ -740,23 +752,35 @@ pimcore.bundle.perspectiveeditor.PerspectiveEditor = class {
...
+            documents: 'pimcore_icon_document',
+            assets: 'pimcore_icon_asset',
+            objects: 'pimcore_icon_object',
+            customview: 'pimcore_icon_custom_views',
+        }
+        const addPerspectiveEditorElementTreeIcon = new CustomEvent(pimcore.events.addPerspectiveEditorElementTreeIcon);
+        addPerspectiveEditorElementTreeIcon.data = elementTreeIcons;
+        document.dispatchEvent(addPerspectiveEditorElementTreeIcon);
+        let elementTreeSettingsFormId = 'perspective_element_tree_settings_form';
+        let elementTreeSettingsForm = new Ext.form.Panel({
```
Comment (robertSt7) [inline src/Resources/public/js/pimcore/perspective/perspective.js]:
```suggestion
        const elementTreeSettingsFormId = 'perspective_element_tree_settings_form';
        const elementTreeSettingsForm = new Ext.form.Panel({
```

File: src/Resources/public/js/pimcore/perspective/perspective.js [hunk 3]
```diff
@@ -765,21 +789,12 @@ pimcore.bundle.perspectiveeditor.PerspectiveEditor = class {
...
-                                documents: 'pimcore_icon_document',
-                                assets: 'pimcore_icon_asset',
-                                objects: 'pimcore_icon_object',
-                                customview: 'pimcore_icon_custom_views',
-                            }
-
-                            record.data.text = newValue;
-                            record.data.iconCls = iconCls[newValue];
+                            Ext.each(elementTreeSettingsForm.query('[name^=perspective_editor]'), function(component) {
+                                let componentName = 'perspective_editor_'+newValue;
```
Comment (robertSt7) [inline src/Resources/public/js/pimcore/perspective/perspective.js]:
```suggestion
                                let componentName = 'perspective_editor_' + newValue;
```

File: src/Resources/public/js/pimcore/perspective/perspective.js [hunk 4]
```diff
@@ -765,21 +789,12 @@ pimcore.bundle.perspectiveeditor.PerspectiveEditor = class {
...
+                                component.setHidden(component.name == componentName ? false : true);
```
Comment (robertSt7) [inline src/Resources/public/js/pimcore/perspective/perspective.js]:
```suggestion
                                component.setHidden(component.name !== componentName);
```

**Pull request merged**